### PR TITLE
Use Rails caching for current_event

### PR DESCRIPTION
**Pull request by jcoyne**
Move the caching of current_event and schedule into the controller layer
This allows you to use a cache and have all processes return the same
values (unless you're using memory store).

### CODE REVIEW

**Comment by caseyhelbling** [comment]
I'm dragging my feet on this because I want to give it a little closer look. I'm gonna go through the other PRs and come back to this one... I hope that doesn't screw up merging stuff..

File: src/app/controllers/schedules_controller.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
 class SchedulesController < ApplicationController
   def index
-    unless Settings.show_schedule?
+    unless Settings.show_schedule? || params[:force]
```
Comment (caseyhelbling) [inline src/app/controllers/schedules_controller.rb]: It seems the param `force` really means `dont_use_the_cached_event` ?

File: src/app/controllers/schedules_controller.rb [hunk 2]
```diff
@@ -55,4 +55,22 @@ def ical
     render :text => cal.to_ical, :content_type => 'text/calendar'
   end
 
+  protected
+
+  def schedule(event)
+    Rails.cache.fetch("#{event.cache_key}/schedule") do
```
Comment (caseyhelbling) [inline src/app/controllers/schedules_controller.rb]: Does the behavior of `event.cache_key` refresh/change once the event changes too? Is that why we don't need to pass the params `force` into this method too?

File: src/app/controllers/schedules_controller.rb (hunk 1 above)
Comment (jcoyne) [inline src/app/controllers/schedules_controller.rb]: right. it allowed us to preview changes prior to the cache expiring.

File: src/app/controllers/schedules_controller.rb (hunk 2 above)
Comment (jcoyne) [inline src/app/controllers/schedules_controller.rb]: Yes. Precisely

**Pull request merged**